tests/extensions/config/resources/test_modifying_db_config.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=missing-docstring
+import json
+
+
+def test_modifying_db_config_by_admin(flask_app_client, admin_user, db):
```

Comment:
Looks good but while it's not required for this PR, I'd say there needs to be a lot more testing here, invalid user, incorrect password, missing password etc.
This test is quite a long function, I think it would be worth looking at factoring some of the blocks out into separate helper functions that could be called for multiple tests, and then these could have parameters, eg the user for the patch operations

Reply:
Howard, I have overhauled how the REST API tests were performed by adding a lot more checks and permissions.

Reply:
Looks very extensive. Unfortunately the changes have caused the assets tests that follow then to fail but if the asset tests are run in isolation they pass. I'll be interested to know what the fix is for this

Reply:
It looks like most of the reasons our tests were failing has to do with the new database import persisting across the session, some state expectations for submission tests that needed to be removed, and the fact that some database folders did not live on disk (see new `_ensure_storage` function)

To highlight test state issues, I've added an automated `invoke app.db._reset` and three random pytest calls (back to back) to the automated testing on GitHub to check for these.